File: 프로그래머스/2/42860. 조이스틱/조이스틱.java
class Solution {
    public int solution(String name) {
        int len = name.length();
        int answer = 0;
        
        // 알파벳 변경 횟수 계산
        for (int i = 0; i < len; i++) {
            char c = name.charAt(i);
            int move = Math.min(c - 'A', 'Z' - c + 1);
            answer += move;
        }

        // 커서 이동 최소화 계산
        int minMove = len - 1; // 오른쪽으로만 이동할 경우

        for (int i = 0; i < len; i++) {
            int next = i + 1;
            while (next < len && name.charAt(next) == 'A') {
                next++;
            }
            int move = i + len - next + Math.min(i, len - next);
            minMove = Math.min(minMove, move);
        }

        answer += minMove;
        return answer;
    }
}

File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    n = len(name)
    answer = 0
    
    
    for char in name:
        
        diff = ord(char) - ord('A')
        answer += min(diff, 26 - diff)

   
    min_move = n - 1 
    
    for i in range(n):
        
        next_i = i + 1
        while next_i < n and name[next_i] == 'A':
            next_i += 1
            
        distance = i + (n - next_i) + min(i, n - next_i)
        min_move = min(min_move, distance)

    answer += min_move
    return answer